Allow Weapons without SlotNumber

edthebat opened the pull request:
If a weapon has no SlotNumber, check all existing playerclasses, see if one of them has the weapon in a WeaponSlot, and use that number.

Note: No work has been done on SlotPriority yet...

edthebat committed:
Update WeaponSwitch.txt

If a weapon has no SlotNumber, check all existing playerclasses, see if one of them has the weapon in a WeaponSlot, and use that number.

Note: No work has been done on SlotPriority yet...

MajorCooke commented:
There needs to be a toggleable option to switch between my version and this one too. Otherwise, this will throw priorities out of the window.

edthebat commented:
That depends on if you want to accept this as-is. I could, alternatively, rework this to accommodate priorities as well. Should I assume, at that point, you would want to just replace your methods with mine? Assuming there are no other functional regressions, of course...

MajorCooke commented:
Not going to merge it right now.

Changing it shouldn't be difficult, just passing in the playerinfo and checking the cvar much like how I did the InputProcess should suffice perfectly.

I request this change simply because there will be weapons that I want to disable via the usage of a slot number outside the range, such as if I want to make a weapon unselectable on my end.

Furthermore, all weapons found that don't have a slot priority assigned to them must go behind every other categorized weapon, if it's not already. I haven't checked that one out yet.

I'll simply wait for you to make the changes first.

edthebat commented:
Understood. I'll keep working until it meets these criteria.

Currently any weapon that still has no slot (even after checking playerclasses) will be marked as Slot -1. I haven't fully tested how that will play out in-game.

edthebat committed:
Include SlotPriority check

Adding to previous commit, the slot priority of a weapon is now captured as well.

Testing is positive so far, but I think I would like to ask for some sample cases to test against, to make sure all scenarios are accounted for.

edthebat commented:
Added a second commit that includes capturing the slot priority of the weapon.

See commit comments re: test cases for added certainty/clarity

MajorCooke commented:
I have yet to see an option toggle appear so it only navigates through weapons with defined slot priorities and numbers.

This is a requirement. I don't want to select some weapons that I purposely don't assign any slots or priorities towards.

edthebat commented:
Still not understanding the need, so I'm closing this for now.
